### PR TITLE
[project-base][SSP-2614] - refresh usermenu styling

### PR DESCRIPTION
**Pull request by TomasGottvald**
#### Description, the reason for the PR
Refresh styling for logged user, refresh Menu Iconic.

#### Fixes issues
... 

#### Check the appropriate checkboxes below, please

- [ ] [BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/) 
- [ ] New feature 
- [x] I have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)

## Summary by CodeRabbit

- **New Features**
	- Enhanced visual testing logic by allowing offsets for specific elements in the cart and payment components.
	- Streamlined snapshot comparison behavior across various test cases, improving the accuracy of visual regression tests.
	- Introduced a mechanism to wait for a stable DOM before executing actions in order tests, enhancing test reliability.
	- Improved user interface on the customer page with new icons for actions and refined styling for better usability.

- **Bug Fixes**
	- Improved error handling and response validation in the user registration and promo code application processes, enhancing robustness and preventing potential runtime errors.
	- Adjusted test configurations to eliminate retries during test runs, providing immediate feedback on failures.

----
:globe_with_meridians: Live Preview:
  - https://tg-ssp-2614-usermenu.odin.shopsys.cloud
  - https://cz.tg-ssp-2614-usermenu.odin.shopsys.cloud

### CODE REVIEW

File: project-base/storefront/components/Layout/Header/MenuIconic/MenuIconicElements.tsx [hunk 1]
```diff
@@ -11,24 +11,26 @@ export const MenuIconicItem: FC<{ title?: string }> = ({ children, className, ti
 
 type MenuIconicItemLinkProps = { onClick?: () => void; href?: string; title?: string; type?: PageType };
 
+const menuIconicSubItemLink = 'flex items-center h-14 px-5 py-3 text-sm text-text no-underline font-semibold';
```
Comment (chlebektomas) [inline project-base/storefront/components/Layout/Header/MenuIconic/MenuIconicElements.tsx]: Could you please change name similar to `menuIconicItemLinkTwClass`

**Review by chlebektomas — CHANGES_REQUESTED** [review]

File: project-base/storefront/components/Layout/Header/MenuIconic/MenuIconicElements.tsx (hunk 1 above)
Comment (TomasGottvald) [reply project-base/storefront/components/Layout/Header/MenuIconic/MenuIconicElements.tsx]: This classname is already used on line 32. 
But you probably want me to add TwClass on the end of classname. So I did (;

Comment (chlebektomas) [reply project-base/storefront/components/Layout/Header/MenuIconic/MenuIconicElements.tsx]: exactly!

File: project-base/storefront/components/Basic/Icon/UserIcon.tsx [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
         <path
-            d="M255.992 315.584c-59.93 0-111.747 14.714-152.185 45.576C63.368 392.022 35.587 438.08 20 495.766l50.822 13.732c13.368-49.474 35.166-83.777 64.924-106.488 29.758-22.711 68.752-34.781 120.246-34.781 51.492 0 90.487 12.074 120.246 34.789 29.76 22.714 51.558 57.021 64.926 106.482l50.822-13.736C476.4 438.092 448.62 392.035 408.182 361.17c-40.437-30.865-92.257-45.586-152.19-45.586zm0-313.084c-72.376 0-131.613 59.237-131.613 131.613 0 72.377 59.237 131.616 131.613 131.616 72.377 0 131.615-59.24 131.615-131.616S328.37 2.5 255.992 2.5zm0 52.646c43.925 0 78.969 35.043 78.969 78.967 0 43.925-35.044 78.969-78.969 78.969s-78.967-35.044-78.967-78.969c0-43.924 35.042-78.967 78.967-78.967z"
-            fill="currentColor"
-            overflow="visible"
+            d="M12.5 14.788a5 5 0 1 0 0-10 5 5 0 0 0 0 10Zm0 0c-4.418 0-8 2.686-8 6m8-6c4.418 0 8 2.686 8 6"
+            stroke="currentColor"
+            strokeLinecap="round"
+            strokeWidth="2"
         />
     </svg>
```
Comment (chlebektomas) [inline project-base/storefront/components/Basic/Icon/UserIcon.tsx:9]: I noticed that the user icon appears similar to the others. Could you please double-check this?
image

File: project-base/storefront/components/Layout/Header/MenuIconic/MenuIconicItemUserAuthenticated.tsx [hunk 3]
```diff
@@ -16,50 +23,104 @@ export const MenuIconicItemUserAuthenticated: FC = () => {
         ['/customer', '/customer/orders', '/customer/edit-profile'],
         url,
     );
+    const [isClicked, setIsClicked] = useState(false);
+    const [isHovered, setIsHovered] = useState(false);
```
Comment (chlebektomas) [inline project-base/storefront/components/Layout/Header/MenuIconic/MenuIconicItemUserAuthenticated.tsx:30]: I would suggest adding a delay when the user hovers over the icon, similar to how `const isMenuOpenedDelayed = useDebounce(isMenuOpened, 200);` is implemented in `NavigationItem.tsx`

**Review by chlebektomas — CHANGES_REQUESTED** [review]

File: project-base/storefront/components/Basic/Icon/UserIcon.tsx (hunk 2 above)
Comment (TomasGottvald) [reply project-base/storefront/components/Basic/Icon/UserIcon.tsx:9]: True fixed.
obrazek

File: project-base/storefront/components/Layout/Header/MenuIconic/MenuIconicItemUserAuthenticated.tsx (hunk 3 above)
Comment (TomasGottvald) [reply project-base/storefront/components/Layout/Header/MenuIconic/MenuIconicItemUserAuthenticated.tsx:30]: used useDebounce, that pushed me to add check if hover is used on desktop or not. There were ability to hover with mouse on developer respo tool. Thx

**Review by chlebektomas — APPROVED** [review]

**Pull request merged**